themes/default/content/blog/stack-reference-output-details/index.md

```diff
@@ -0,0 +1,100 @@
+---
+title: "Streamline access to stack outputs with OutputDetails"
+date: 2023-03-24T17:24:02-07:00
+meta_desc: >-
+  You can now read and transform outputs from stack references
+  without calling Output.apply.
+meta_image: meta.png
+authors: [abhinav-gupta]
+tags: [features, stack-reference]
+---
+
+You can use the new `StackReference.getOutputDetails` method
+to get outputs from other stacks directly&mdash;without calling `Output.apply`.
+
+<!--more-->
+
+## What are Stack References?
+
+> If you already understand stack references,
+> skip on to [Introducing `OutputDetails`](#introducing-outputdetails).
+
+Stack references are a useful feature of Pulumi
+that enables you to access output values from one stack in another.
+For instance,
+you may have a stack that provisions infrastructure resources such as a VPC
+and a separate stack that deploys applications to that VPC.
+
+Let's say you have a 'network' stack that provisions an AWS VPC,
+and an 'app' stack that deploys your application on an EC2 instance in that VPC.
+The app stack can reference the output of the network stack
+to obtain the VPC ID, which it needs to configure the EC2 instance.
+
+```typescript
+const network = new pulumi.StackReference("acmecorp/network/other");
+const vpcId = network.getOutput("vpcId");
+```
+
+The `getOutput` method above returns an `Output` value that holds the result.
+You can then call `Output.apply` to perform transformations on the output value.
+For example,
+if you need to extract a subnet ID from a JSON string in the output,
+you can use the following code:
+
+```typescript
+const subnetConfig = network.getOutput("subnetConfig");
+const subnetId = subnetConfig.apply(output => JSON.parse(output).subnetId);
+```
+
+The above works but is limited in that
+the output value obtained from the `apply` transformation
+is also an `Output` value.
+This means that accessing or manipulating this value directly is not possible,
+and you must continue to use `apply` repeatedly to consume the value.
```

Comment:
Probably not in this section, but somewhere we should probably explain that these methods are only exposed for stack references because stack references _always_ return known values. Without that explanation I can see people asking "well why do I need apply everywhere else if these outputs don't need it", and the answer is "unknown" values.

Reply:
Good point. I thought about a "why" but wasn't sure whether it wasn't a premature explanation.
A second data point here proves that we need it.

Reply:
Added a section for that.